### ISSUE TITLE
A permission to allow teachers to see when their classes are scheduled.

### ISSUE THREAD

**Issue by benjaminjkraft** [learning-unlimited/ESP-Website#480]
Sometimes during scheduling we want teachers to not be able to see where their classes are, because we're still moving them around.  Right now that's only possible by closing the entire teacherreg mainpage.  It would be nice if there were a seperate permission/userbit to show it.

**Comment by pushkarkumar06** [comment]
Hi @willgearty ,

I’d like to work on this issue if it’s still open.

From the description, it seems the goal is to introduce a permission or userbit that allows teachers to see when their classes are scheduled, while still giving administrators the flexibility to hide this information during scheduling changes.

My initial idea would be to explore adding a dedicated permission check that controls whether teachers can view scheduling details on the teacher registration page, without requiring the entire page to be disabled.

If this approach sounds reasonable, I’d be happy to investigate the relevant parts of the codebase and work on a solution.

Please let me know if I can take this up.

Thanks!

**Comment by kkbrum** [comment]
@pushkarkumar06 As you currently have 6 open issues assigned, I’ll hold off on assigning you in case others want to take a stab.

**Comment by Gourigupta2004** [comment]
Hi, I am backend developer and have experience with Django. I would like to work on this issue, please assign this to me.

**Comment by Gourigupta2004** [comment]
Hi, I am a backend developer and have experience with Django. I would like to work on this issue, please assign this to me.

**Comment by Gourigupta2004** [comment]
Hi @kkbrum @willgearty, 

I was previously working on another issue related to Learning Unlimited and have now started working on this issue.

I’m currently exploring the scheduling flow and relevant models/APIs for teachers. I’ll share updates soon and raise a PR once the implementation is ready.

**Comment by kkbrum** [comment]
Great! No need to comment / tag us unless you have any questions.

**Comment by willgearty** [comment]
@Gourigupta2004 any update on this?

**Comment by Gourigupta2004** [comment]
Hi @willgearty, 
Yes, I will be raising a PR by tomorrow.

**Comment by Gourigupta2004** [comment]
Hi @willgearty @kkbrum,

I have implemented the feature and opened a PR for the same. Kindly review it at your convenience and share your feedback.

Thank you!

**Comment by Gourigupta2004** [comment]
Hi @willgearty @kkbrum, It's been a really long time since I opened a PR for this issue. It's is yet to be reviewed or merged. Kindly share you feedback on the same. Other reviewers have already approved the PR. 

Thanks

**Comment by willgearty** [comment]
We've been slowly working our way through the PR backlog. We'll get to yours in time. Thanks for your contribution and for your patience.

**Comment by Gourigupta2004** [comment]
Thanks @willgearty
